### PR TITLE
Use Write-Log and RemoteLogCollector.

### PR DESCRIPTION
**Pull request by m-kostrzewa**
Puts detailed logs into detailed_logs directory (using `Write-Log` function and `RemoteLogCollector`).

Just look at it: http://logs.opencontrail.org/winci/github/WinContrail/contrail-win-ci-gh/1364/detailed_logs/

http://logs.opencontrail.org/winci/github/WinContrail/contrail-win-ci-gh/1364/detailed_logs/Single%20compute%20node%20protocol%20tests%20with%20utils.TCP%20connection%20works.log

### CODE REVIEW

**Comment by m-kostrzewa** [comment]
recheck no bug

**Comment by krdln** [comment]
Perhaps let's import `PesterLogger` and thus `Write-Log` explicitely and change `Write-Log` to just check whether a `Write-LogImpl` is registered, if not, falling back to `Write-Host`?

**Comment by m-kostrzewa** [comment]
Just look at it: http://logs.opencontrail.org/winci/github/WinContrail/contrail-win-ci-gh/1364/detailed_logs/

**Comment by m-kostrzewa** [comment]
> Perhaps let's import PesterLogger and thus Write-Log explicitely and change Write-Log to just check whether a Write-LogImpl is registered, if not, falling back to Write-Host?
@krdln good idea, but I really want to get this merged first, since it's been hanging in limbo for weeks.

**Comment by krdln** [comment]
[There are still some `Write-Host`s that are not captured](http://logs.opencontrail.org/winci/github/WinContrail/contrail-win-ci-gh/1364/log.txt.gz#_2018-04-18_06_25_18_642000)

File: CIScripts/Test/PesterLogger/PesterLogger.ps1 [hunk 1]
```diff
@@ -3,7 +3,8 @@
 . $PSScriptRoot/Get-CurrentPesterScope.ps1
 
 function Initialize-PesterLogger {
-    Param([Parameter(Mandatory = $true)] [string] $Outdir)
+    Param([Parameter(Mandatory = $true)] [string] $Outdir,
+          [Parameter(Mandatory = $false)] [PSSessionT[]] $Sessions)
```
Comment (jacek-izykowski) [inline CIScripts/Test/PesterLogger/PesterLogger.ps1]: Why do we need it?

**Comment by jacek-izykowski** [comment]
What about Tests/Protocol/TunnellingWithAgent.Tests.ps1?

File: CIScripts/Test/Tests/Protocol/SingleHostWithUtils.Tests.ps1 [hunk 2]
```diff
@@ -151,6 +154,8 @@ Describe "Single compute node protocol tests with utils" {
             $ContrailNM.RemoveNetwork($ContrailNetwork)
             Remove-Variable "ContrailNetwork"
         }
+
+        Merge-Logs -LogSources @($DriverLogs, $AgentLogs)
```
Comment (jacek-izykowski) [inline CIScripts/Test/Tests/Protocol/SingleHostWithUtils.Tests.ps1]: What's going to happen when some earlier step in this block fails? Can we manage to handle merging logs anyway?

File: CIScripts/Test/Tests/Protocol/SingleHostWithUtils.Tests.ps1 [hunk 3]
```diff
@@ -166,6 +171,16 @@ Describe "Single compute node protocol tests with utils" {
...
         $IisTcpTestDockerImage = "iis-tcptest"
 
+        [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSUseDeclaredVarsMoreThanAssignments",
+            "AgentLogs", Justification="Analyzer doesn't understand relation of Pester blocks"
+        )]
+        $AgentLogs = New-LogSource -Sessions $Session -Path "C:/ProgramData/Contrail/var/log/contrail/*.log"
+        [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSUseDeclaredVarsMoreThanAssignments",
+            "DriverLogs", Justification="Analyzer doesn't understand relation of Pester blocks"
+        )]
+        $DriverLogs = New-LogSource -Sessions $Session -Path "C:/ProgramData/ContrailDockerDriver/log.txt"
```
Comment (jacek-izykowski) [inline CIScripts/Test/Tests/Protocol/SingleHostWithUtils.Tests.ps1]: If we move those variable initialization close to the place, where variables are actually used, we might get rid of static analysis message suppression.

**Comment by m-kostrzewa** [comment]
recheck no bug

**Comment by m-kostrzewa** [comment]
recheck no bug

**Comment by m-kostrzewa** [comment]
recheck no bug

**Comment by m-kostrzewa** [comment]
recheck no bug

File: CIScripts/Test/Tests/Integration/DockerDriver.Tests.ps1 [hunk 4]
```diff
@@ -83,6 +79,7 @@ Describe "Docker Driver" {
     }
 
     AfterEach {
-        Merge-Logs -LogSources $DriverLogs
+        $Logs = New-LogSource -Sessions $Session -Path "C:/ProgramData/Contrail/var/log/contrail/*.log"
```
Comment (jacek-izykowski) [inline CIScripts/Test/Tests/Integration/DockerDriver.Tests.ps1]: Can we do anything with this constant that appears in many tests?

File: CIScripts/Test/PesterLogger/RemoteLogCollector.ps1 [hunk 5]
```diff
@@ -12,6 +12,12 @@ function New-LogSource {
     }
 }
 
+function New-ContrailLogsSource {
+    Param([Parameter(Mandatory = $true)] [PSSessionT[]] $Sessions)
+
+    New-LogSource -Sessions $Sessions -Path "C:/ProgramData/Contrail/var/log/contrail/*.log"
```
Comment (m-kostrzewa) [inline CIScripts/Test/PesterLogger/RemoteLogCollector.ps1]: Hm.. I think I disagree with this diff. We shouldn't put anything related to log paths in a helper module. We want it to be pure.

If anything, this function should be in `Testbed.ps1` or some other file in `Testenv/` directory.

Comment (m-kostrzewa) [reply CIScripts/Test/PesterLogger/RemoteLogCollector.ps1]: @jacek-izykowski what do you think

**Comment by krdln** [comment]
recheck no bug

**Comment by krdln** [comment]
recheck no bug

**Pull request merged**